### PR TITLE
E2E: Transfer Precompiles and Child Records Checks

### PR DESCRIPTION
**Pull request by georgiyazovaliiski**
**Description**:
This PR adds:
1. E2E scenarios for the transferTokens, transferNft, transferNfts.
2. ChildRecords checks in the HAPI testing suite.

**Checklist**

- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-node/settings.txt [hunk 1]
```diff
@@ -6,7 +6,6 @@ dbRestore.active,                              1
 doUpnp,                                        false
 maxOutgoingSyncs,                              1
 reconnect.active,                              1
-reconnect.asyncInputStreamTimeoutMilliseconds, 60000
```
Comment (tinker-michaelj) [inline hedera-node/settings.txt:9]: This property has been renamed to,
```
reconnect.asyncStreamTimeoutMilliseconds, 60000
```

**Review by tinker-michaelj — APPROVED** [review]
LGTM so far!

File: test-clients/src/main/java/com/hedera/services/bdd/spec/assertions/ChildRecordsAsserts.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+import java.lang.reflect.Method;
+import java.util.List;
+import java.util.function.Function;
+
+import static com.hedera.services.bdd.spec.assertions.EqualityAssertsProviderFactory.shouldBe;
+import static com.hedera.services.bdd.spec.assertions.EqualityAssertsProviderFactory.shouldNotBe;
+import static java.util.Collections.EMPTY_LIST;
+
+public class ChildRecordsAsserts extends TransactionRecordAsserts {
```
Comment (stoqnkpL) [inline test-clients/src/main/java/com/hedera/services/bdd/spec/assertions/ChildRecordsAsserts.java:44]: Do we need this class at all?

File: hedera-node/settings.txt (hunk 1 above)
Comment (stoqnkpL) [inline hedera-node/settings.txt:9]: Why do we need to remove this?

File: test-clients/src/main/java/com/hedera/services/bdd/spec/queries/meta/HapiGetTxnRecord.java [hunk 3]
```diff
@@ -286,6 +293,24 @@ private void assertPriority(HapiApiSpec spec, TransactionRecord actualRecord) th
 		expectedDebits.ifPresent(debits -> assertEquals(debits, asDebits(actualRecord.getTransferList())));
 	}
 
+	private void assertChildRecords(HapiApiSpec spec, List<TransactionRecord> actualRecords) throws Throwable {
+		if (childRecordsExpectations.isPresent()) {
+			final var expectedChildRecords = childRecordsExpectations.get();
+
+			assertEquals(expectedChildRecords.size(), actualRecords.size(), "Expected and actual child records should" +
+					" be the same size.");
```
Comment (stoqnkpL) [inline test-clients/src/main/java/com/hedera/services/bdd/spec/queries/meta/HapiGetTxnRecord.java:301]:
```suggestion
```suggestion
assertEquals(expectedChildRecords.size(), actualRecords.size(), 
					String.format("Expected %d child records, got %d", expectedChildRecords.size(), actualRecords.size()));
```
```

File: test-clients/src/main/java/com/hedera/services/bdd/spec/queries/meta/HapiGetTxnRecord.java [hunk 4]
```diff
@@ -337,12 +362,24 @@ private void assertTopicRunningHash(HapiApiSpec spec, TransactionRecord actualRe
 		}
 	}
 
+	private void assertTransferLists(HapiApiSpec spec, TransactionRecord childRecord) {
+
+	}
+
```
Comment (stoqnkpL) [inline test-clients/src/main/java/com/hedera/services/bdd/spec/queries/meta/HapiGetTxnRecord.java:368]: we should delete this

**Review by stoqnkpL — CHANGES_REQUESTED** [review]

**Review by stoqnkpL — APPROVED** [review]
LGTM!

**Review by tinker-michaelj — APPROVED** [review]
LGTM!!

**Pull request merged**